### PR TITLE
[PBM-577] Support choice of S3 storage class

### PR DESCRIPTION
**Pull request by dalbani**
https://jira.percona.com/projects/PBM/issues/PBM-577

### CODE REVIEW

File: pbm/storage/s3/s3.go [hunk 1]
```diff
@@ -118,6 +119,10 @@ func New(opts Conf, l *log.Event) (*S3, error) {
 const defaultPartSize = 10 * 1024 * 1024 // 10Mb
 
 func (s *S3) Save(name string, data io.Reader, sizeb int) error {
+	storageClass := s.opts.StorageClass
+	if storageClass == "" {
+		storageClass = s3.StorageClassStandard
+	}
```
Comment (dAdAbird) [inline pbm/storage/s3/s3.go:125]: Although storage class is related only to upload (S3.Save method) I'd like to move this check to [the Cast() method](https://github.com/percona/percona-backup-mongodb/blob/bb6286a6cbeb749d32b62b29aca84b4da54f09d6/pbm/storage/s3/s3.go#L56). This method is called during config set so no need to run checks on each save. Besides, it'd be more sound with the rest of the code.

**Review by dAdAbird — CHANGES_REQUESTED** [review]
Hi @dalbani 
Thanks for the contribution!
See my comment.

File: pbm/storage/s3/s3.go (hunk 1 above)
Comment (dalbani) [reply pbm/storage/s3/s3.go:125]: Good suggestion  @dAdAbird, I'll change that.

Comment (dalbani) [reply pbm/storage/s3/s3.go]: Fix in latest commit.

**Comment by dAdAbird** [comment]
@dalbani can you please resolve git conflicts with the current main (https://github.com/percona/percona-backup-mongodb/pull/613 remrge aftermath)? Thanks!

**Comment by dalbani** [comment]
Rebased.

**Comment by dalbani** [comment]
I should probably document the new `storage.s3.storageClass` property I suppose?

**Comment by dAdAbird** [comment]
> I should probably document the new `storage.s3.storageClass` property I suppose?

Would be highly appreciated.
If not, no problem, we can add docs later during the release.

**Comment by dalbani** [comment]
I have added a small blurb in the documentation.

**Comment by dAdAbird** [comment]
Thanks @dalbani

**Comment by dalbani** [comment]
Now that `MaxUploadParts` and `StorageClass` have been added to the `Conf` struct, I'd like to use them in PSMDB operator.
I suppose it requires a new release of PBM and an upgrade of the dependencies, just like it was just done for 1.6.1?

**Comment by dalbani** [comment]
What I meant in my last comment is what I implemented in https://github.com/percona/percona-server-mongodb-operator/pull/831.

**Pull request merged**